ui/src/modules/bfla/index.js

```diff
@@ -0,0 +1,124 @@
+import React, {useState} from 'react';
+import {useFetch} from 'hooks';
+import Loader from 'components/Loader';
+import Button from 'components/Button';
+import TitleValueDisplay, { TitleValueDisplayRow } from 'components/TitleValueDisplay';
+import BflaStatusIcon, {BFLA_STATUS_TYPES_MAP} from './BflaStatusIcon';
+import BflaModal from './BflaModal';
+import MODULE_TYPES from '../MODULE_TYPES.js';
+import bflaApiInventory from './BflaApiInventory';
+
+import './bfla.scss';
+
+const BflaPlugin = (props) => {
+    const {eventId} = props;
+    const [{ loading, data: bflaData }, updateBflaEvent] = useFetch(`modules/bfla/event/${eventId}`);
+    const [showBflaModal, setShowBflaModal] = useState();
+
+    if (loading) {
+        return <Loader />;
+    }
+
+    if (!bflaData) {
+        return <div>No BFLA data found</div>;
+    }
+
+    const {bflaStatus, sourceK8sObject, destinationK8sObject, external} = bflaData;
+    const sourceName = external ? 'EXTERNAL' : sourceK8sObject.name;
+    const sourceKind = sourceK8sObject ? sourceK8sObject.kind : '';
+    const destinationName = destinationK8sObject ? destinationK8sObject.name : '';
+    const destinationKind = destinationK8sObject ? destinationK8sObject.kind : '';
+
+    return (
+        <div>
+            <TitleValueDisplayRow>
+                <TitleValueDisplay title={external ? "Source" : "Source Name"}>{sourceName}</TitleValueDisplay>
+                {!external &&
+                    <TitleValueDisplay title="Source Kind">{sourceKind}</TitleValueDisplay>
+                }
+                <TitleValueDisplay title="Destination Name">{destinationName}</TitleValueDisplay>
+                <TitleValueDisplay title="Destination Kind">{destinationKind}</TitleValueDisplay>
+            </TitleValueDisplayRow>
+            <TitleValueDisplayRow>
+                <BflaStatus bflaStatus={bflaStatus} sourceName={sourceName} />
+            </TitleValueDisplayRow>
+            <TitleValueDisplayRow>
+                {(bflaStatus !== BFLA_STATUS_TYPES_MAP.LEARNING.value && bflaStatus !== BFLA_STATUS_TYPES_MAP.NO_SPEC.value) &&
+                    <Button onClick={() => setShowBflaModal(true)} >{bflaStatus === BFLA_STATUS_TYPES_MAP.LEGITIMATE.value ? "Mark as Illegitimate" : "Mark as Legitimate"}</Button>
+                }
+            </TitleValueDisplayRow>
+
+            {showBflaModal &&
+                <BflaModal
+                    eventId={eventId}
+                    type={bflaStatus === BFLA_STATUS_TYPES_MAP.LEGITIMATE.value ? 'deny' : 'approve' }
```

Comment:
Maybe add a const for "deny" and "approve". The strings seem to be repeating on the code